src/PaymentGateways/PayPalCommerce/PayPalOrder.php

```diff
@@ -0,0 +1,84 @@
+<?php
+namespace Give\PaymentGateways\PayPalCommerce;
+
+use Give\Helpers\ArrayDataSet;
+use InvalidArgumentException;
+use stdClass;
+
+/**
+ * Class PayPalOrder
+ * @package Give\PaymentGateways\PayPalCommerce
+ *
+ * @property-read string $id
+ * @property-read string $intent
+ * @property-read stdClass $purchaseUnit
+ * @property-read stdClass $payer
+ * @property-read string $createTime
+ * @property-read string $updateTime
+ * @property-read string $links
+ * @property-read string $status
+ */
+class PayPalOrder {
+	/**
+	 * Create PayPalOrder object from given array.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param $array
+	 *
+	 * @return PayPalOrder
+	 */
+	public static function fromArray( $array ) {
+		$order = new static();
+
+		$order->validate( $array );
+		$array = ArrayDataSet::camelCaseKeys( $array );
+
+		foreach ( $array as $itemName => $value ) {
+			if ( 'purchaseUnits' === $itemName ) {
+				$value = current( $value );
+			}
+
+			$order->{$itemName} = $value;
+		}
+
+		return $order;
+	}
+
+	/**
+	 * Get payment details for PayPal order.
+	 *
+	 * @since 2.8.0
+	 *
+	 *
+	 * @return PayPalPayment
+	 */
+	public function getPayment() {
+		if ( property_exists( $this->purchaseUnits, 'payments' ) ) {
+			$payments = $this->purchaseUnits->payments;
+
+			if ( property_exists( $payments, 'captures' ) ) {
+				return PayPalPayment::fromArray( (array) current( $payments->captures ) );
+			}
+		}
+
+		return null;
```

Comment:
We want to avoid these kinds of conditions, as we never really know the intended shape of our class by looking at it. You'd have to read through where `fromArray` is used and these methods to even get an idea of the shape.

Reply:
fixed: https://github.com/impress-org/givewp/compare/f549508bd6a0...ae1fb915985b